Wedge bond from atrop error

tadhurst-cdd opened the pull request:
<!--
Thanks for contributing a pull request! 
-->
#### Reference Issue
fixed a bug in determining which bonds to wedge for an atropisomer


#### What does this implement/fix? Explain your changes.
If a bond had directions set to help determine the cis-trans config for a double bond, and also COULD be used to define an atropisomer bond, an error was thrown.  Now the error is not generated and the bond is simply skipped.

#### Any other comments?

Also made a change to help my build work having to do with string_view inside a catch2 macro.

greglandrum commented on Code/GraphMol/StructChecker/StripSmallFragments.cpp:
This change is unrelated to the PR. Please remove it

tadhurst-cdd replied:
Again, I have removed the change.  But is the structure checker flag is on for the build, this causes an error.

greglandrum reviewed (changes requested):
The changes which are related to the topic of PR look fine to me.
The two changes that are not related ought to be removed unless there's a reason behind them that I'm not seeing.

tadhurst-cdd commented on Code/GraphMol/FileParsers/file_parsers_catch.cpp:
```diff
@@ -6319,7 +6319,7 @@ TEST_CASE("MaeWriter basic testing", "[mae][MaeWriter][writer]") {
     std::string_view ctBlock(&mae[ctBlockStart]);
 
-    CHECK(ctBlock == MaeWriter::getText(*mol));
+    CHECK((ctBlock == MaeWriter::getText(*mol)) == true);
```
I will remove the change, but without it the system does not build.  There is a problem between string_view and Catch2.

greglandrum replied:
> I will remove the change, but without it the system does not build. There is a problem between string_view and Catch2.

This does not happen for me on any machine I build the RDKit on, nor does it happen on any of the CI systems. This is likely a problem with your local configuration

DavidACosgrove replied:
This happened to me also, on my Macs.  It seems that it is because Catch2 needs to be compiled with C++17 compatibility in order to use string_view and by default its build uses C++14.  I deleted my installation in /usr/local and RDKit built a version that worked for me.

greglandrum reviewed (approved):
LGTM

tadhurst-cdd commented:
I just made a separate PR for this change.  I think we are using the structure checker at CDD

 

Dr. Tad Hurst

Collaborative Drug Discovery, Inc.

 ***@***.***> ***@***.***

314-610-4313

 

From: Greg Landrum ***@***.***> 
Sent: Friday, May 3, 2024 10:01 AM
To: rdkit/rdkit ***@***.***>
Cc: tadhurst-cdd ***@***.***>; Author ***@***.***>
Subject: Re: [rdkit/rdkit] Wedge bond from atrop error (PR #7355)

 

@greglandrum commented on this pull request.

  _____  

On Code/GraphMol/StructChecker/StripSmallFragments.cpp <https://github.com/rdkit/rdkit/pull/7355#discussion_r1589328782> :

Again, I have removed the change. But is the structure checker flag is on for the build, this causes an error.

Thanks for pointing that out. I can take a look at that or you could submit a separate PR with that change; I'm trying to keep individual PRs as focused as possible

General comment: I don't use the structure checker code and we don't build it as part of the CI or (I think) any of the public builds. It ought to be deprecated and just removed at some point.

—
Reply to this email directly, view it on GitHub <https://github.com/rdkit/rdkit/pull/7355#discussion_r1589328782> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/A22KQE2MGTGKSQEOJGSRPFLZAORDFAVCNFSM6AAAAABGEKPHMSVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDAMZYGM2TINZZHA> .
You are receiving this because you authored the thread.  <https://github.com/notifications/beacon/A22KQE5ET2FH35NZO4YAC6LZAORDFA5CNFSM6AAAAABGEKPHMSWGG33NNVSW45C7OR4XAZNRKB2WY3CSMVYXKZLTORJGK5TJMV32UY3PNVWWK3TUL5UWJTTZP3JW4.gif> Message ID: ***@***.*** ***@***.***> >
